code/modules/flufftext/Hallucination.dm

```diff
@@ -702,307 +747,367 @@ Gunshots/explosions/opening doors/less rare audio (done)
-				var/list/possible_points = list()
-				for(var/turf/open/floor/F in view(src,world.view))
-					possible_points += F
-				if(possible_points.len)
-					var/turf/open/floor/target = pick(possible_points)
-
-					switch(rand(1,4))
-						if(1)
-							//to_chat(src, "Space")
-							halimage = image('icons/turf/space.dmi',target,"[rand(1,25)]",TURF_LAYER)
-						if(2)
-							//to_chat(src, "Lava")
-							halimage = image('icons/turf/floors/lava.dmi',target,"smooth",TURF_LAYER)
-						if(3)
-							//to_chat(src, "Chasm")
-							halimage = image('icons/turf/floors/Chasms.dmi',target,"smooth",TURF_LAYER)
-						if(4)
-							//to_chat(src, "C4")
-							halimage = image('icons/obj/grenade.dmi',target,"plastic-explosive2",OBJ_LAYER+0.01)
-
-
-					if(client) client.images += halimage
-					sleep(rand(40,60)) //Only seen for a brief moment.
-					if(client) client.images -= halimage
-					halimage = null
-		if("death")
-			//Fake death
-			hal_screwyhud = SCREWYHUD_DEAD
-			SetUnconscious(400)
-			var/area/area = get_area(src)
-			to_chat(src, "<span class='deadsay'><b>[mind.name]</b> has died at <b>[area.name]</b>.</span>")
-			if(prob(50))
-				var/list/dead_people = list()
-				for(var/mob/dead/observer/G in GLOB.player_list)
-					dead_people += G
-				var/mob/dead/observer/fakemob = pick(dead_people)
-				if(fakemob)
-					sleep(rand(30, 60))
-					to_chat(src, "<span class='deadsay'><b>DEAD: [fakemob.name]</b> says, \"[pick("rip","welcome [first_name()]","you too?","is the AI malf?",\
-					 "i[prob(50)?" fucking":""] hate [pick("blood cult", "clock cult", "revenants", "abductors","double agents","viruses","badmins","you")]")]\"</span>")
-			sleep(rand(50,70))
-			hal_screwyhud = SCREWYHUD_NONE
-			SetSleeping(0)
-		if("husks")
-			if(!halbody)
-				var/list/possible_points = list()
-				for(var/turf/open/floor/F in view(src,world.view))
-					possible_points += F
-				if(possible_points.len)
-					var/turf/open/floor/target = pick(possible_points)
-					switch(rand(1,4))
-						if(1)
-							var/image/body = image('icons/mob/human.dmi',target,"husk",TURF_LAYER)
-							var/matrix/M = matrix()
-							M.Turn(90)
-							body.transform = M
-							halbody = body
-						if(2,3)
-							halbody = image('icons/mob/human.dmi',target,"husk",TURF_LAYER)
-						if(4)
-							halbody = image('icons/mob/alien.dmi',target,"alienother",TURF_LAYER)
-
-					if(client) client.images += halbody
-					spawn(rand(30,50)) //Only seen for a brief moment.
-						if(client) client.images -= halbody
-						halbody = null
+					target.halimage = image('icons/effects/effects.dmi',danger_point,"electricity2",OBJ_LAYER+0.01)
+
+
+			if(target.client)
+				target.client.images += target.halimage
+			sleep(rand(200,450))
+			if(target.client)
+				target.client.images -= target.halimage
+			QDEL_NULL(target.halimage)
+	qdel(src)
+
+/datum/hallucination/death
+	cost = 40
+
+/datum/hallucination/death/New(mob/living/carbon/T, forced = TRUE)
+	set waitfor = 0
```

Comment:
Because calling new /datum/hallucination/death apparently waits for the end of the New() proc before continuing, without removing waitfor, and it was causing bugs with the deathbolt hallucination